### DISCUSSION TITLE
Can't run in exclusive full screen(EFS) since "Minimise osu!when switching to another app" version update

### DISCUSSION THREAD

**Discussion by Xeladoes** [ppy/osu#27446] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Can't run in exclusive full screen since(EFS) "Minimise osu!when switching to another app" version update, and the game gets unstable with all kinds of lag\latency problems.
**I'm describing the problem in the following order.**
**1. The devices with the problem**
**2. How I notice the problem & problem description**
**3. Some EFS problem behavier**
**4. How to turn EFS on unstably**
**5. My modification to settings**

**1. The devices with the problem**
I have 2 devices, one OMEN 9 top(13900HX+4060 ver)(I spend 2 month to modify settings with disgusting gaming experience that destroyed my game sense and I have a discusstion about that says the problem was fixed which is actually not(lag fixed but the latency is not), and I'm so tired about that so I'm not talking about it now, is going to be another issue), other one is the desktop we are facing with now. Desktop hardware are following:
System: Windows 10 Enterprise Edition 22H2 (19045.4046)
CPU: AMD Ryzen7 5800X 8-Core Processor
motherboard: MSI MAGB550m MORTAR WIFI(MS-7C94)
GPU: NVIDIA GeForce RTX3070 Vulcan OC(8GB/Colorful)
2 monitors: H27T22S[SKG2725] (using 2k 144Hz) + VA2762-2K-HD[VSC7B3D] (using 2k 60Hz, sometimes closed if games are laggy)
memory: 16GB*2 DDR4 3600MHz
harddrive: Samsung SSD 970 EVO 1TB
sound card: ESI MAYA44 eX

**2. How I notice the problem**
Just 2 weeks ago I played with the old version with the smooth experience for a half hour and because I have the disgusting experience with the top, I checked the game is on EFS and I checked the settings before update.
I updated lazer and I knew something bad is about to happen, then I notice the game is not on EFS because I know there is a new setting called "Minimise osu!when switching to another app" is going to be here to make EFS off, then for sure I open the game and experienced the lag(random lag and continuous lag) and input latency immediately. In the following week I tried everything I can to modify with the settings(talk about that in 5. My modification to settings) to make it better, it is just better but not better then before(is still not on EFS), still some random lag and sometimes latency.

**3. Some EFS problem behavier**
The way to see EFS is doing alt+Tab switch and if the window switched immediately that meas EFS is not on, even if the game says you are on EFS and guaranteed no input latency(by using OpenGL), it is not(I mostly use Dx11).

**4. How to turn EFS on unstably**
I found two only ways to use EFS when I'm writing this:
1. open lazer with 2 screens on, the game will running in main monitor, and change monitor in game, then the EFS will be on but the refresh rate will follow the setting of the secondary screen(60Hz x4 is 240 but it will run on 300Hz and 600Hz for x8, weird, I checked my monitor it is running on 60Hz). And the game is still laggy sometimes.
2. open lazer with only main monitor on, after the game is running, press win+P and switch to extended mode to turn other monitor on, then the game will be on EFS with main monitor refresh rate, haven't try this yet, hopefully the solution, but you will need two monitors.

**5. My modification to settings**
Before I show the settings, I need carify that I've tested **EVERY settings** in a control variables way, which is play a few round of games before a new specific setting changed, and then another few round of games to see the differences. My current settings are already the optimal solution after testing many different settings now. This is really **A LOT OF WORK(more then 2 month)**, but only in this way can we accurately determine whether a certain setting is effective, so I have to do it, or I'll quit the game. And I quit the game already, because I'm tired with all this lag, I don't care about the improvment, my passion for this game has burned out after all this.

- In game: Full screen + Default resolution(2560*1440) + "Minimise osu!when switching to another app" on + Dx11 + 4x refresh rate + Multithreaded + use hardware acc
- Drivers are download from MSI MAGB550m MORTAR WIFI motherboard official site, only graphic drivers are always up to date using NVIDIA Gefore Experience. Motherboard BIOS was updated to the newest.
- Bitlocker off
- Ultimate Performance Power Plan
- MSI Center not installed
- Use task manager to find the game file and open "Disable full screen optimization" in file properties
- 10 GB Virtual Memory
- In bios I only opened PBO2 and only overclocked memory to 3800hz, ECO off
- NVIDIA Control Panel use 3D settings and for osu is following (sorry it's in Chinese), GSYNC off(not supported), Phyx use GPU
![image](https://github.com/ppy/osu/assets/79356888/aa9a5a87-c08a-4e5b-b9b4-07809a5907fc)
![image](https://github.com/ppy/osu/assets/79356888/57f022cb-7b8e-4950-bf05-553e18a74d93)
![image](https://github.com/ppy/osu/assets/79356888/ac677da3-d751-4b92-8848-45019982b850)

After update lazer I did follow modification that reduced a lot of lag\latency. But remember, before lazer update the game was running perfectly, and all other games are doing good too, I didn't have to do all this.
**BIOS settings modification:**
- Close all security settings including fTPM
- Close CPPC

**system modification:**
- Close High precision clock by running bcdedit /deletevalue useplatformclock and bcdedit /set useplatformclock false in cmd

### Version

Lazer 2024.221.0

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14457401/compressed-logs.zip)

**Comment by bdach** [comment]
The setting in question doesn't do anything to "disable exclusive fullscreen"? In fact it is only shown to the user _in_ exclusive fullscreen?

**Reply by Xeladoes** [reply]
![image](https://github.com/ppy/osu/assets/79356888/e5a76729-ffb1-42c8-b42a-a3b1c48bd2d7)

**Reply by Xeladoes** [reply]
In this snipshot I can use dc and snipshot

**Comment by SupDos** [comment]
Also on Win10 here, and the DirectX renderer has never been able to switch into Exclusive Full screen for me, only the OpenGL renderer does. Since it sounds like you're mainly using DirectX, try the other renderer. Sometimes I've had to restart my PC when Windows decides to force optimizations even on OpenGL though, so make sure you do that as well. Also, 

>The way to see EFS is doing alt+Tab switch and if the window switched immediately

Is it possible that it's just switching so fast you think it's not Exclusive Full screen? osu!'s switch is extremely fast for me, only barely being noticeable by a black screen appearing for a microsecond (blink and you can miss it). The way to 100% confirm if exclusive full screen is working is to change your Windows volume (or bring up the Windows Volume HUD) while the game is focused. If you don't see it, then you're in exclusive full screen.

**Reply by Xeladoes** [reply]
No, it's always not on EFS when I start the game, for now I'm using "4. How to turn EFS on unstably" method 2nd

**Comment by Xeladoes** [comment, marked as answer]
I don't what want great work you did but these problem doesn't exist anymore, thanks!

And also the saying in my discussion is not correct for the game version now, you can just ignore them:
> 3. Some EFS problem behavier
> The way to see EFS is doing alt+Tab switch and if the window switched immediately that meas EFS is not on, even if the game says you are > on EFS and guaranteed no input latency(by using OpenGL), it is not(I mostly use Dx11).
